File: js/solve/238_Product_of_Array_Except_Self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    var temp = [];
    var product = 1;
    for(var i=0; i<nums.length; i++){
        temp[i] = product;
        product *= nums[i];
    }
    console.log(temp)
    product = 1;
    for(var i=nums.length-1; i>=0; i--){
        temp[i] *= product;
        product *= nums[i];
    }
    return temp;
};

File: 238.product-of-array-except-self.py
#
# @lc app=leetcode id=238 lang=python3
#
# [238] Product of Array Except Self
#

# @lc code=start
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result = []
        product = 1
        for i in range(len(nums)):
            result.append(product)
            product *= nums[i]
        product = 1
        for i in reversed(range(len(nums))):
            result[i] *= product
            product *= nums[i]
        return result
# @lc code=end

